[BUG]

# If you do not follow this template, your issue may be ignored or delayed due to lack of information.

**Describe the bug**
```
Feb 20 16:27:11 meshmonitor meshmonitor[3303]: [ERROR] [DatabaseService] Failed to insert telemetry: Error: Failed query: insert into "telemetry" ("id", "nodeId", "nodeNum", "telemetryType", "timestamp", "value", "unit", "createdAt", "packetTimestamp", "packetId", "channel", "precisionBits", "gpsAccuracy") values (default, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
Feb 20 16:27:11 meshmonitor meshmonitor[3303]: params: !ec356669,3962922601,snr_local,1771604831177,-17.25,dB,1771604831177,,4290698168,,,
```

**To Reproduce**
Version 3.6.3, running on Proxmox LXC image, connected to Postgresql, fresh install, fresh database.

**Expected behavior**
See a constant stream of failures to input telemetry data into the database, maybe due to date format?


**Environment (please complete the following information):**
* Host OS: Linux proxmox LXC (debian12)
* Deployment type: Docker? Mac App? Windows App? or Baremetal? 
  * if Docker, which Container version are you using?
  * If the app, what host OS are you using?
  * If NPM, which commit?
* Database backend: PostGreSQL

**Additional context**
```
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: [INFO] 🔐 Received NodeInfo with public key for !29560841 (AB4YY-T114): rOH7jz0lDQZ1deAsTEv0... (32 bytes)
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: [ERROR] [DatabaseService] Failed to insert telemetry: Error: Failed query: insert into "telemetry" ("id", "nodeId", "nodeNum", "telemetryType", "timestamp", "value", "unit", "createdAt", "packetTimestamp", "packetId", "channel", "precisionBits", "gpsAccuracy") values (default, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: params: !29560841,693504065,messageHops,1771605244507,5,hops,1771605244507,,2848684734,,,
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: [ERROR] [DatabaseService] Failed to insert telemetry: Error: Failed query: insert into "telemetry" ("id", "nodeId", "nodeNum", "telemetryType", "timestamp", "value", "unit", "createdAt", "packetTimestamp", "packetId", "channel", "precisionBits", "gpsAccuracy") values (default, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: params: !29560841,693504065,snr_local,1771605244507,-12.5,dB,1771605244507,,2848684734,,,
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: [ERROR] [DatabaseService] Failed to insert telemetry: Error: Failed query: insert into "telemetry" ("id", "nodeId", "nodeNum", "telemetryType", "timestamp", "value", "unit", "createdAt", "packetTimestamp", "packetId", "channel", "precisionBits", "gpsAccuracy") values (default, $1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)
Feb 20 16:34:04 meshmonitor meshmonitor[3303]: params: !29560841,693504065,rssi,1771605244507,-119,dBm,1771605244507,,2848684734,,,
Feb 20 16:34:21 meshmonitor meshmonitor[3303]: [INFO] 📊 LocalStats request packet created: 38 bytes for local node !d4a70ce8, packetId=4217107865, requestId=3715769804
Feb 20 16:34:21 meshmonitor meshmonitor[3303]: [INFO] 📤 LocalStats request sent to local node !d4a70ce8
```

--- a/src/db/schema/mysql-create.ts
+++ b/src/db/schema/mysql-create.ts
@@ -115,7 +115,7 @@ export const MYSQL_SCHEMA_SQL = `
     unit VARCHAR(255),
     createdAt BIGINT NOT NULL,
     packetTimestamp BIGINT,
-    packetId INT,
+    packetId BIGINT,
     channel INT,
     precisionBits INT,
     gpsAccuracy DOUBLE,

--- a/src/db/schema/postgres-create.ts
+++ b/src/db/schema/postgres-create.ts
@@ -110,7 +110,7 @@ export const POSTGRES_SCHEMA_SQL = `
     unit TEXT,
     "createdAt" BIGINT NOT NULL,
     "packetTimestamp" BIGINT,
-    "packetId" INTEGER,
+    "packetId" BIGINT,
     channel INTEGER,
     "precisionBits" INTEGER,
     "gpsAccuracy" DOUBLE PRECISION

--- a/src/db/schema/telemetry.ts
+++ b/src/db/schema/telemetry.ts
@@ -37,7 +37,7 @@ export const telemetryPostgres = pgTable('telemetry', {
   unit: pgText('unit'),
   createdAt: pgBigint('createdAt', { mode: 'number' }).notNull(),
   packetTimestamp: pgBigint('packetTimestamp', { mode: 'number' }),
-  packetId: pgInteger('packetId'),
+  packetId: pgBigint('packetId', { mode: 'number' }),
   // Position precision tracking metadata
   channel: pgInteger('channel'),
   precisionBits: pgInteger('precisionBits'),
@@ -55,7 +55,7 @@ export const telemetryMysql = mysqlTable('telemetry', {
   unit: myVarchar('unit', { length: 32 }),
   createdAt: myBigint('createdAt', { mode: 'number' }).notNull(),
   packetTimestamp: myBigint('packetTimestamp', { mode: 'number' }),
-  packetId: myInt('packetId'),
+  packetId: myBigint('packetId', { mode: 'number' }),
   // Position precision tracking metadata
   channel: myInt('channel'),
   precisionBits: myInt('precisionBits'),

--- a/src/server/migrations/075_upgrade_telemetry_packetid_bigint.ts
+++ b/src/server/migrations/075_upgrade_telemetry_packetid_bigint.ts
@@ -1,0 +1,78 @@
+/**
+ * Migration 075: Upgrade telemetry packetId from INTEGER to BIGINT
+ *
+ * Meshtastic packet IDs are unsigned 32-bit values (up to ~4.3 billion),
+ * but PostgreSQL/MySQL INTEGER is signed 32-bit (max 2,147,483,647).
+ * This causes insert failures for packetId values above INT4 max.
+ *
+ * SQLite is unaffected because its INTEGER type is 64-bit.
+ */
+
+import type { Database } from 'better-sqlite3';
+import { logger } from '../../utils/logger.js';
+
+/**
+ * SQLite migration: No changes needed
+ * SQLite INTEGER is already 64-bit
+ */
+export const migration = {
+  up: (_db: Database): void => {
+    logger.debug('Migration 075: SQLite INTEGER is already 64-bit, no changes needed');
+  },
+
+  down: (_db: Database): void => {
+    logger.debug('Migration 075 down: No changes needed for SQLite');
+  }
+};
+
+/**
+ * PostgreSQL migration: Upgrade packetId from INTEGER to BIGINT
+ */
+export async function runMigration075Postgres(client: import('pg').PoolClient): Promise<void> {
+  logger.info('Running migration 075 (PostgreSQL): Upgrading telemetry.packetId to BIGINT...');
+
+  try {
+    const result = await client.query(`
+      SELECT column_name FROM information_schema.columns
+      WHERE table_name = 'telemetry' AND column_name = 'packetId'
+    `);
+
+    if (result.rows.length > 0) {
+      await client.query(`ALTER TABLE telemetry ALTER COLUMN "packetId" TYPE BIGINT`);
+      logger.debug('Upgraded telemetry.packetId to BIGINT');
+    } else {
+      logger.debug('Column telemetry.packetId does not exist, skipping');
+    }
+  } catch (error: any) {
+    logger.error('Failed to upgrade telemetry.packetId:', error.message);
+    throw error;
+  }
+
+  logger.info('Migration 075 complete (PostgreSQL): telemetry.packetId upgraded to BIGINT');
+}
+
+/**
+ * MySQL migration: Upgrade packetId from INT to BIGINT
+ */
+export async function runMigration075Mysql(pool: import('mysql2/promise').Pool): Promise<void> {
+  logger.info('Running migration 075 (MySQL): Upgrading telemetry.packetId to BIGINT...');
+
+  try {
+    const [rows] = await pool.query(`
+      SELECT COLUMN_NAME FROM information_schema.COLUMNS
+      WHERE TABLE_NAME = 'telemetry' AND COLUMN_NAME = 'packetId'
+    `);
+
+    if (Array.isArray(rows) && rows.length > 0) {
+      await pool.query(`ALTER TABLE telemetry MODIFY COLUMN packetId BIGINT`);
+      logger.debug('Upgraded telemetry.packetId to BIGINT');
+    } else {
+      logger.debug('Column telemetry.packetId does not exist, skipping');
+    }
+  } catch (error: any) {
+    logger.error('Failed to upgrade telemetry.packetId:', error.message);
+    throw error;
+  }
+
+  logger.info('Migration 075 complete (MySQL): telemetry.packetId upgraded to BIGINT');
+}

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -79,6 +79,7 @@ import { migration as meshcorePermissionMigration, runMigration071Postgres, runM
 import { migration as dmUnreadIndexMigration, runMigration072Postgres, runMigration072Mysql } from '../server/migrations/072_add_messages_dm_unread_index.js';
 import { migration as packetIdMigration, runMigration073Postgres, runMigration073Mysql } from '../server/migrations/073_add_packet_id_to_telemetry.js';
 import { migration as showMeshCoreNodesMigration, runMigration074Postgres, runMigration074Mysql } from '../server/migrations/074_add_show_meshcore_nodes_preference.js';
+import { migration as telemetryPacketIdBigintMigration, runMigration075Postgres, runMigration075Mysql } from '../server/migrations/075_upgrade_telemetry_packetid_bigint.js';
 import { validateThemeDefinition as validateTheme } from '../utils/themeValidation.js';
 
 // Drizzle ORM imports for dual-database support
@@ -952,6 +953,7 @@ class DatabaseService {
     this.runDmUnreadIndexMigration();
     this.runPacketIdMigration();
     this.runShowMeshCoreNodesMigration();
+    this.runTelemetryPacketIdBigintMigration();
     this.ensureAutomationDefaults();
     this.warmupCaches();
     this.isInitialized = true;
@@ -2416,6 +2418,25 @@ class DatabaseService {
       logger.debug('show_meshcore_nodes migration completed successfully');
     } catch (error) {
       logger.error('Failed to run show_meshcore_nodes migration:', error);
+      throw error;
+    }
+  }
+
+  private runTelemetryPacketIdBigintMigration(): void {
+    const migrationKey = 'migration_075_telemetry_packetid_bigint';
+    try {
+      const migrationStatus = this.getSetting(migrationKey);
+      if (migrationStatus === 'completed') {
+        logger.debug('Migration 075 (telemetry_packetid_bigint) already completed');
+        return;
+      }
+
+      logger.debug('Running migration 075: Upgrade telemetry packetId to BIGINT...');
+      telemetryPacketIdBigintMigration.up(this.db);
+      this.setSetting(migrationKey, 'completed');
+      logger.debug('telemetry packetId bigint migration completed successfully');
+    } catch (error) {
+      logger.error('Failed to run telemetry packetId bigint migration:', error);
       throw error;
     }
   }
@@ -10853,6 +10874,8 @@ class DatabaseService {
       // Run migration 074: Add show_meshcore_nodes to user_map_preferences
       await runMigration074Postgres(client);
 
+      // Run migration 075: Upgrade telemetry packetId to BIGINT
+      await runMigration075Postgres(client);
 
       // Verify all expected tables exist
       const result = await client.query(`
@@ -10988,6 +11011,8 @@ class DatabaseService {
       // Run migration 074: Add show_meshcore_nodes to user_map_preferences
       await runMigration074Mysql(pool);
 
+      // Run migration 075: Upgrade telemetry packetId to BIGINT
+      await runMigration075Mysql(pool);
 
       // Verify all expected tables exist
       const [rows] = await connection.query(`

